lib/naughty/core.lua

```diff
@@ -380,6 +399,14 @@ local function cleanup(self, reason)
         n.idx = k
     end
 
+    -- Remove from the global active list.
+    for k, n in ipairs(naughty._active) do
+         if n == self then
+            table.remove(naughty._active, k)
+            naughty.emit_signal("property::active")
+         end
+    end
```

Comment:
The logic was that the number of active notification never gets that large. Also keeping a inverted table with only the notification as key and index as value would require to update the most of the list entries every time a notification gets destroyed anyway. This would not be fully linear since some long running notification (index) will eventually bubble up which would mean their indices would remain stable for longer. However I am not sure this is worth the extra complexity.

Keeping track of the index for each "position" as the old implementation did ended up causing half of the regressions we had so far. This could be modified if it ever becomes a problem, but right not I believe using other strategies wont be faster. If some day it becomes common to have hundreds of active notifications (smartphone style), then the rational will no longer be valid.